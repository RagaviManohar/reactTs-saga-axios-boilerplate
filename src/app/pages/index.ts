//
// HomePage
//
export { default as HomePage } from './HomePage/Loadable';
//
// LoginPage
//
export { default as LoginPage } from './LoginPage/Loadable';
