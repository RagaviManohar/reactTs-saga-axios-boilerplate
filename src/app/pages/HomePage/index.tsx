/**
 * HomePage
 */

import { memo } from 'react';

import { HelmetWrapper } from 'app/components';

export const HomePage = () => {
  return (
    <>
      <HelmetWrapper title="Home" description="Home Page" />
      <div>
        <p>Home Page</p>
      </div>
    </>
  );
};

export default memo(HomePage);
