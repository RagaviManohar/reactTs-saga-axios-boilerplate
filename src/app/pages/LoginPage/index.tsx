/**
 * LoginPage
 */

import { memo } from 'react';

import { HelmetWrapper } from 'app/components';

export const LoginPage = () => {
  return (
    <>
      <HelmetWrapper title="Login" description="Login Page" />
      <div>
        <p>Login Page</p>
      </div>
    </>
  );
};

export default memo(LoginPage);
