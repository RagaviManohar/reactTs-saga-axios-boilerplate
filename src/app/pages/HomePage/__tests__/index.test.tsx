import { render, cleanup, HelmetProvider } from 'jest.setup';

import { HomePage } from 'app/pages/HomePage';

afterEach(cleanup);

const renderComponent = ({ ...props }) =>
  render(
    <HelmetProvider>
      <HomePage {...props} />
    </HelmetProvider>,
  );

describe('<HomePage/>', () => {
  it('expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderComponent({});
    expect(spy).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { container } = renderComponent({});
    expect(container.firstChild).toMatchSnapshot();
  });
});
