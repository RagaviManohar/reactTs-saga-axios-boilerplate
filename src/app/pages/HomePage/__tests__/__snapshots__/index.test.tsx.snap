// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage/> should match snapshot 1`] = `
<div>
  <p>
    Home Page
  </p>
</div>
`;
